ktor-http/common/src/io/ktor/http/HeaderValueWithParameters.kt

```diff
@@ -93,7 +93,27 @@ private fun String.checkNeedEscape(): Boolean {
     return false
 }
 
-private fun String.isQuoted(): Boolean = length > 1 && first() == '\"' && last() == '\"'
+private fun String.isQuoted(): Boolean {
+    if (length < 2) {
+        return false
+    }
+    if (first() != '"' || last() != '"') {
+        return false
+    }
+    var startIndex = 1
+    do {
+        val index = indexOf('"', startIndex)
+        if (index == -1 || index == lastIndex) {
+            break
+        }
+        if (this[index - 1] != '\\') {
```

Comment:
This is not guaranteed because the slash could be escaped as well: `\\"` I personally don't think that it's so serious here. Do you?

Reply:
thanks, updated and added test